### ISSUE TITLE
how to parse a buffer?

### ISSUE THREAD

**Issue by mderazon** [wdavidw/node-csv-parse#57]
Trying to parse a Buffer into csv with auto columns detection

``` js
csv.parse(req.file.buffer, { columns: true }, function (err, data) {
 // outputs `Error: Invalid data argument`
})
```

Is there a way to parse Buffers without transforming them to strings first ?

**Comment by wdavidw** [comment]
We could accept buffers but internally they will be converted to strings

**Comment by mderazon** [comment]
I see, so I guess there's no real benefit other than cosmetic.

**Comment by wdavidw** [comment]
you're right

**Comment by karlpokus** [comment]
I don't understand. ELI5? I have the exact same dilemma. Parsing a buffer without the options object works (and returns arrays with arrays) but not if I supply it.

**Comment by wdavidw** [comment]
You mean that this work `csv.parse(req.file.buffer, { columns: true }, function (err, data) {}` while this doesnt work `csv.parse(req.file.buffer, function (err, data) {}`?

**Comment by karlpokus** [comment]
Hi! @wdavidw No. Quite the opposite. Supplying an options object *does not work*. I'm using csv-parse btw.
